contracts/axelarnet-gateway/src/contract/query.rs

```diff
@@ -0,0 +1,168 @@
+use axelar_wasm_std::FnExt;
+use cosmwasm_std::Deps;
+use router_api::{CrossChainId, Message};
+
+use crate::state::{self, MessageWithStatus};
+
+pub fn sent_messages(deps: Deps, cc_ids: Vec<CrossChainId>) -> Result<Vec<Message>, state::Error> {
+    cc_ids
+        .into_iter()
+        .map(|cc_id| {
+            state::may_load_sent_msg(deps.storage, &cc_id)?
+                .ok_or(state::Error::MessageNotFound(cc_id))
+        })
+        .collect::<Result<Vec<_>, _>>()?
+        .then(Ok)
```

Comment:
```suggestion
        .collect::<Result<Vec<_>, _>>()
```
Does this work?

Reply:
ah, yes, i had changed the error type, so this should work now